pinot-core/src/main/java/org/apache/pinot/core/indexsegment/generator/SegmentGeneratorConfig.java

```diff
@@ -382,11 +382,7 @@ public void setSegmentNamePostfix(String postfix) {
   }
 
   public String getTimeColumnName() {
-    if (_segmentTimeColumnName != null) {
-      return _segmentTimeColumnName;
-    }
-    // TODO: if segmentTimeColumnName is null, getQualifyingFields DATETIME. If multiple found, throw exception "must specify primary timeColumnName"
```

Comment:
@npawar  can you confirm that it is ok to remove this and keep backward comaptibility?

Reply:
Here getting TIME field is an no-op. Moved the TODO for DATETIME field support into `setSchema()`

Reply:
this looks fine to me.